src/ui/AccessibleLinksFormatter.cpp

```diff
@@ -0,0 +1,170 @@
+   {
+      S.AddFixedText(mMessage);
+      return;
+   }
+
+   wxString translated = mMessage.Translation();
+
+   ProcessArguments(translated);
+
+   const int borderSize = S.GetBorder();
+
+   S.Prop(0).StartPanel();
+   S.StartWrapLay();
+   {
+      size_t currentPosition = 0;
+
+      S.SetBorder(0);
+
+      if (borderSize > 0)
+         S.AddSpace(borderSize);
+
+      for (const FormatArgument& argument : mFormatArguments)
+      {
+         if (argument.PlaceholderPosition == wxString::npos)
+             // ProcessArguments sorts the mFormatArguments
+             // So only wxString::npos can occur after wxString::npos
+             break; 
+
+         // Add everything between currentPosition and PlaceholderPosition
+
+         if (currentPosition != argument.PlaceholderPosition)
+         {
+            const size_t substrLength =
+               argument.PlaceholderPosition - currentPosition;
+
+            S.Prop(0).AddFixedText(
+               Verbatim(translated.substr(currentPosition, substrLength)));
+         }
+
+         // Add hyperlink
+
+         wxHyperlinkCtrl* hyperlink = safenew wxHyperlinkCtrl(
+            S.GetParent(), wxID_ANY, argument.Value.Translation(),
+            argument.TargetURL);
+
+         if (argument.Handler)
+         {
+            hyperlink->Bind(
+               wxEVT_HYPERLINK, [handler = argument.Handler](wxHyperlinkEvent& evt) { 
+                  handler();
+            });
+         }
+
+         S.AddWindow(hyperlink, wxALIGN_TOP | wxALIGN_LEFT);
+
+         // Update the currentPostion to the first symbol after the Placeholder
+
+         currentPosition =
+            OffsetPosition(argument.PlaceholderPosition, argument.Placeholder.Length());
+
+         if (currentPosition >= translated.Length())
+            break;
+      }
+
+      if (currentPosition < translated.Length())
+         S.AddFixedText(Verbatim(translated.substr(currentPosition)));
+   }
+   S.EndWrapLay();
+   S.EndPanel();
+
+   S.SetBorder(borderSize);
+}
+
+void AccessibleLinksFormatter::ProcessArguments(wxString translatedMessage)
+{
+   // Arguments with the same placeholder are processed left-to-right.
+   // Lets track the last known position of the placeholder
+   std::unordered_map<wxString, size_t> knownPlaceholderPosition;
+
+   for (FormatArgument& argument : mFormatArguments)
+   {
+      auto it = knownPlaceholderPosition.find(argument.Placeholder);
+
+      const size_t startingPosition =
+         it != knownPlaceholderPosition.end() ?
+            OffsetPosition(it->second, argument.Placeholder.length()) :
+            0;
+
+      const size_t placeholderPosition =
+         startingPosition == wxString::npos ?
+            wxString::npos :
+            translatedMessage.find(argument.Placeholder, startingPosition);
+
+      argument.PlaceholderPosition = placeholderPosition;
+      knownPlaceholderPosition[argument.Placeholder] = placeholderPosition;
+   }
+
+   std::stable_sort(
```

Comment:
Why stable sort?  How could two FormatArguments have the same PlaceholderPosition now?  Maybe if one placeholder is a prefix of another?  Hm, sounds like a caller mistake then.

Reply:
This is an edge case handling. I'm not sure, that this edge case will ever occur.

Technically it is possible, that a placeholder is missing from the translation to one language, but is present in another (why? I don't know).

In this case, `placeholderPosition` will be `npos`. If we have several cases of that - `stable_sort` will preserve the original order for the format arguments.

So I think, that for 99.9% of cases `sort` is sufficient. 

But for the majority of cases, both `sort` and `stable_sort` will be no-op, because there will be only one item to format.

Reply:
These is a way for msgfmt to make errors if the format substitutions don’t match.  Let’s make sure to use it when accepting translations.

Reply:
I'm not quite sure what I should here.

BTW stable_sort is replaced by sort as a part of addressing another comment

Reply:
My last comment mentioned something I should remember in the next round of acceptance of translations, not anything for you to do now.

If English has %s but some .po file omits it, I think there is a way to detect that error when I accept translations.